lib/ts/recipe/emailverification/emailVerificationClaim.ts

```diff
@@ -31,10 +31,19 @@ export class EmailVerificationClaimClass extends BooleanClaim {
 
         this.validators = {
             ...this.validators,
-            isVerified: (refetchTimeOnFalseInSeconds: number = 10, maxAgeInSeconds: number = 300) => ({
+            isVerified: (refetchTimeOnFalseInSeconds: number = 10, maxAgeInSeconds?: number) => ({
                 ...this.validators.hasValue(true, maxAgeInSeconds),
                 shouldRefetch: (payload, userContext) => {
                     const value = this.getValueFromPayload(payload, userContext);
+
+                    // Avoid refetching by default if maxAgeInSeconds is undefined and the value is true.
+                    if (maxAgeInSeconds === undefined && value === true) {
+                        return false;
```

Comment:
since you added an if statement to return early in one condition I'd prefer if you restructured the existing return statement to match it (i.e.: expand the logic expression into multiple ifs to match the new addition)